revvy/robot_config.py

```diff
@@ -72,6 +72,21 @@ def dict_get_first(dictionary, keys):
     return None
 
 
+def make_script_name_common(script_idx, assignment_type, detail):
+    return 'script_{}_{}_{}'.format(
+        script_idx, assignment_type, detail)
+
+
+def make_analog_script_name(analog, script_idx):
+    detail = 'channels_' + '_'.join(map(str, analog['channels']))
+    return make_script_name_common(script_idx, 'analog', detail)
+
+
+def make_button_script_name(script_idx, button_idx):
+    return make_script_name_common(script_idx,
+        'button', f'{button_idx}')
```

Comment:
Why is button_idx in an f-string here?

Reply:
By design I wanted to pass last argument to be of type string, though strictly its not needed, but stresses the idea that make_script_name_common accepts string in places where it is called.